secretsstoresynccontroller/api/v1alpha1/secretstoresync_types.go

```diff
@@ -0,0 +1,109 @@
+/*
+Copyright 2023.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
```

Comment:
Use this boilerplace https://github.com/kubernetes-sigs/secrets-store-csi-driver/blob/main/hack/boilerplate.go.txt for license. You can pass it to the `controller-gen` command like this

https://github.com/kubernetes-sigs/secrets-store-csi-driver/blob/ac51f4738d370d54f2e185f70cc27fcaa650bdb6/Makefile#L504